service/history/nDCEventsReapplier.go

```diff
@@ -47,25 +50,22 @@ func newNDCEventsReapplier(
 ) nDCEventsReapplier {
 
 	return &nDCEventsReapplierImpl{
-		metricsClient: metricsClient,
-		logger:        logger,
+		historySerializer: persistence.NewPayloadSerializer(),
```

Comment:
take in the shard object and get the historySerializer from shard?

Reply:
moved the serializer to handler level.